### PR TITLE
RNA components: Add Disconnect Dialog

### PR DESCRIPTION
**Pull request by fgiannar**
Add `DisconnectDialog` RNA component -- A reusable component for disconnecting the Jetpack connection.

## Component `DisconnectDialog`
The `DisconnectDialog` component displays a 'Disconnect' button that, upon clicking, will open a Dialog that presents the user the option to Disconnect their site.
Upon confirming, both site and user are disconnected and the user is presented with a success message along with a "Return to WordPress" button that closes the dialog.
If an error occurs while trying to disconnect, a custmomizable error message will appear.

### Properties
- *apiRoot* - string (required), API root URL.
- *apiNonce* - string (required), API Nonce.
- *title* - string, the dialog title. Defaults to: "Are you sure you want to disconnect?"
- *onDisconnected* - callback, to be called after the user has been successfully disconnected AND clicks the "Return to WordPress" button.
- *onError* - callback, to be called when an error occurs while disconnecting.
- *errorMessage* - string, error message to display when an error occurs while disconnecting. Defaults to: "Failed to disconnect. Please try again."

### Important Notes
It's important to note that the `onDisconnected` callback will not immediately trigger upon receiving a successfull API response. This happens because we want to display a success message to the user first within the `DisconnectDialog`.
If a parent consumer for example, were to update their connection status using this event, the `DisconnectDialog` would be hidden before showing the success message to the user.
Because we made this design decision, and to ensure a non-breaking UX, the `Modal` (see `wordpress/components/modal`) used by the `DisconnectDialog` will not close (as usual) using either ESC key or clicking outside of the Dialog.
This way we ensure that `onDisconnected` will always be called via clicking the "Return to WordPress" button, after successfully disconnecting the site.

### Basic Usage
```jsx
import React, { useCallback } from 'react';
import { DisconnectDialog } from '@automattic/jetpack-connection';

const onDisconnectedCallback = useCallback( () => alert( 'Successfully Disconnected' ) );

<DisconnectDialog
	apiRoot={ APIRoot }
	apiNonce={ APINonce }
	onDisconnected={ onDisconnectedCallback }
>
	<p>
		{ __( 'Jetpack is currently powering multiple products on your site.',
                'jetpack' ) }
		<br/>
		{ __( 'Once you disconnect Jetpack, these will no longer work.',
                'jetpack' ) }
    </p>
</DisconnectDialog>
```

#### Jetpack product discussion
pbuNQi-1w3-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
If you're using a JN site, you need to install [this plugin](https://gist.github.com/sergeymitr/bc005e1b893e778c03e4f43debb91c30) to activate the Connection Manager page.

- Fully connect Jetpack.
- Go to "Tools -> Connection Manager".
- You should see a link to "Disconnect". Click on that link and verify the Disconnect Dialog shows up as displayed below:
Screenshot 2021-07-19 at 5 23 30 PM

- Verify that clicking the "Stay connected" button, closes the dialog.

- Verify that upon clicking the "Disconnect" button, your site will be disconnected and you'll see a success message + "Back to WordPress" button, as displayed below:

Screenshot 2021-07-20 at 10 59 21 AM

- Clicking the "Back to WordPress" button should close the Disconnect Dialog and you should see the `ConnectScreen`, in order to Connect your Account

Extra points for:
- Testing in various browsers ( Chrome, Safari, Edge, Firefox, Opera etc)
- Testing in various viewports
- Testing with an error (eg You can modify the [API endpoint for disconnecting the site](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#L1694) to return an error)

### CODE REVIEW

**Comment by keoshi** [comment]
First off, thanks for the tip in getting this reviewed and your help in doing so @fgiannar !

I've collected a few items below, hopefully this actionable format is helpful (let me know if otherwise).


#### Modal

```diff
 .components-modal__frame {
- border: 1px solid #ddd;
+ border-radius: 4px;
}

.jp-disconnect-dialog__content {
+ padding: 80px;
}

.jp-disconnect-dialog__actions {
- padding: 10px 24px;
+ padding: 10px 48px; 
}
```

#### Typography

```diff
.jp-disconnect-dialog h1 {
+ font-size: var( --font-title-large );
+ font-weight: 600;
}

+ .jp-disconnect-dialog h2 {
+ margin: 0;
+ font-size: var( --font-title-small );
+ font-weight: 400;
+ line-height: 1.2;
+ }

+ .jp-disconnect-dialog__actions p {
+ font-size: var( --font-body );
+ }
```

We would need change the description in the `.jp-disconnect-dialog p` tag to `.jp-disconnect-dialog h2` for the styles above.

#### Buttons

```diff
+ .jp-disconnect-dialog__actions .components-button {
+ height: 40px;
+ font-size: var( --font-body-small );
+ border-radius: 4px;
+ }
```

**Review by leogermani — APPROVED** [review]
Tested with the provided instructions and it works as expected.

Tested in Firefox and Chrome, in multiple viewports. Tested with error message.

File: projects/js-packages/connection/components/disconnect-dialog/style.scss [hunk 1]
```diff
@@ -88,11 +107,11 @@
 
 @media (min-width: 960px){
 	.jp-disconnect-dialog{
-		min-width: 800px;
-	}
-
-	&__success{
-		min-width: auto;
+		min-width: 900px;
```
Comment (keoshi) [inline projects/js-packages/connection/components/disconnect-dialog/style.scss:110]: Nice tweak!

**Review by keoshi — COMMENTED** [review]
Sorry for the delay (post-merge), but this is looking sharp! Thanks for all your hard work, @fgiannar !

**Pull request merged**